pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -706,6 +706,58 @@ public void shutdown() throws PulsarClientException {
         }
     }
 
+    @Override
+    public void forceCloseConnection() {
+        if (this.producers != null) {
```

Comment:
There are few problems with this way of closing the connections. 

`channelInactive()` is supposed to be called by Netty when the connection gets closed. With the current approache the original TCP connection will stay open and the broker will still have registered the producer/consumer as connected. 

The right approach here should be to close all the connections in the `ConnectionPool`. That will be triggering the reconnections for all producers/consumers.

Reply:
@merlimat I'm already fix it.